### ISSUE TITLE
add a new class for realizing Early Stopping

### ISSUE THREAD

**Issue by wubowen416** [tensorflow/tensorflow#33739]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.0 but this is for keras.
- Are you willing to contribute it (Yes/No): yes. I have already realized it and now I am using it.

**Describe the feature and the current behavior/state.**

I want to add a new class for realizing Early Stopping. 

They are based on 3 metrics from paper "Early Stopping - but when? Lutz Prechelt, University Karlsrhe". 

1. **Generalization Loss**(GL). GL can be described as a metric that can measure how much current validation loss exceeds the lowest validation loss. Training will stop as soon as GL exceeds a certain threshold. 

2. **Progress Quotient**(PQ).  It considers training strip of length `k`. Progress means how much was the average training error during the strip larger than the minimum training error during the strip. Quotient means use the quotient of GL and Progress. Thus, training will stop as soon as the PQ exceeds a certain threshold. 

3. **UP**. This is simple. Training will stop when the generalization error increased in `s` successive strips.

Combining these metrics, there are 5 modes for doing early stopping. 1, 2, 3, 1+3, 2+3.

Now the EarlyStopping class of Keras can only stop training based on patience and delta, whereas there are advanced and useful early stopping methods.

**Will this change the current api? How?**

No. Only a new class will be added.

**Who will benefit with this feature?**

Basically all workers using Keras.

**Any Other info.**

This idea comes from others paper, I just realized it. For further info, please refer to the paper.
[early stopping - but when.pdf](https://github.com/tensorflow/tensorflow/files/3774558/early.stopping.-.but.when.pdf)

**Comment by rmothukuru** [comment]
@wubowen416,
Sorry for the delayed response. In your comment you stated:

> Are you willing to contribute it (Yes/No): yes. I have already realized it and now I am using it.

Since you have already realized it and are using it, can you please submit a **`PR`** so that the respective Engineers can review it?

Thanks!
